### PR TITLE
Fix rails 7.1 support and migrate to github actions

### PR DESCRIPTION
**Pull request by bliof-fc**
## Migrate to github actions

Run tests against:
- ruby 2.7
- ruby 3.0
- ruby 3.1
- ruby 3.2
- ruby 3.3

Test with rails - 5.2, 6.0, 6.1, 7.0, 7.1
Test with sinatra - 1.4, 2.2, 3.2, 4.0
Test with sidekiq - 5.1, 6.0, 6.1, 6.2, 6.3, 6.4, 6.5, 7.0, 7.1, 7.x

## Fix for rails 7.1

- lib/loga/railtie.rb is changed to work with rails 7.1

In rails 7.1 https://github.com/rails/rails/blob/v7.1.3/actionpack/lib/action_dispatch/middleware/debug_exceptions.rb#L56
vs rails 7.0 https://github.com/rails/rails/blob/v7.0.8/actionpack/lib/action_dispatch/middleware/debug_exceptions.rb#L54

With rails 7.1 there is also a new `ActiveSupport::BroadcastLogger`, that is currently not in use by `loga` 
https://github.com/rails/rails/blob/v7.1.3/activesupport/lib/active_support/broadcast_logger.rb

### CODE REVIEW

**Comment by fc-infosec-sast** [comment]
<h2>✅ SAST Report</h2>

No issues found. Good job! 💪

**Review by popovm — APPROVED** [review]

**Review by sgerrand — COMMENTED** [review]
👋 Hello, FC crew! Thanks for maintaining this gem. I've added some small non-blocking suggestions.

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        options: >-
+          --health-cmd "redis-cli ping"
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+        ports:
+          - 6379:6379
+    strategy:
+      matrix:
+        ruby-version: ['2.7', '3.0', '3.1', '3.2', 'head']
```
[the remark this answers is not in the document]
Comment (bliof-fc) [reply .github/workflows/build-and-test.yml]: 👋 @sgerrand  Thanks for the suggestions 🍻  

This PR is mainly to renew the pipeline and is already dropping 2.5 and 2.6. I didn't want to change any of the actual code of the gem but there was a bug with rails 7.1

Best will be to upgrade the matrix in a follow up PR, where we can address the rubocop issues as well.

[the remark this answers is not in the document]
Comment (sgerrand) [reply .github/workflows/build-and-test.yml]: Makes sense! 🍻

**Pull request merged**